### PR TITLE
Added List Policies API to SecurityAlerts

### PR DESCRIPTION
**Pull request by t-robalt**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedDatabaseSecurityAlertListByDatabase.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"responses": {
+		"200": {
+			"body": {
+				"value": [
+					{
+						"id": "/subscriptions/00000000-1111-2222-3333-444444444444/resourceGroups/securityalert-6852/providers/Microsoft.Sql/managedInstances/securityalert-2080/databases/testdb",
+						"name": "default",
+						"type": "Microsoft.Sql/managedInstances/databases/securityAlertPolicies",
+						"location": "Japan East",
+						"kind": "V12",
```
Comment (sanjaiganesh) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedDatabaseSecurityAlertListByDatabase.json]: >					 [](start = 0, length = 5)

tabs -to--> spaces

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedServerSecurityAlertListByInstance.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        "api-version": "2017-03-01-preview"
+    },
+    "responses": {
+        "200": {
+            "body": {
+				"value": [
+                    {
+						"id": "/subscriptions/00000000-1111-2222-3333-444444444444/resourceGroups/securityalert-4799/providers/Microsoft.Sql/managedInstances/securityalert-6440",
+						"name": "Default",
+						"type": "Microsoft.Sql/managedInstances/securityAlertPolicies",
```
Comment (sanjaiganesh) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedServerSecurityAlertListByInstance.json]: >"type": [](start = 6, length = 7)

add  location, kind

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedServerSecurityAlertListByInstance.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+						"properties": {
+							"state": "Disabled",
+							"emailAccountAdmins": true,
+							"emailAddresses": [ "test@microsoft.com;user@microsoft.com" ],
+							"disabledAlerts": ["Access_Anomaly"],
```
Comment (sanjaiganesh) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedServerSecurityAlertListByInstance.json]: convert tabs to (4) spaces

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ServerSecurityAlertsListByServer.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    },
+    "responses": {
+        "200": {
+            "body": {
+				"value": [
+                    {
+						"id": "/subscriptions/00000000-1111-2222-3333-444444444444/resourceGroups/securityalert-4799/providers/Microsoft.Sql/servers/securityalert-6440/securityAlertPolicies",
+						"name": "Default",
+						"type": "Microsoft.Sql/servers/securityAlertPolicies",
+						"properties": {
```
Comment (sanjaiganesh) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ServerSecurityAlertsListByServer.json]: add location, kind

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ServerSecurityAlertsListByServer.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+							"state": "Disabled",
+							"emailAccountAdmins": true,
+							"emailAddresses": [ "test@microsoft.com;user@microsoft.com" ],
+							"disabledAlerts": ["Access_Anomaly"],
+							"retentionDays": 0,
+							"storageAccountAccessKey": "",
```
Comment (sanjaiganesh) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ServerSecurityAlertsListByServer.json]: convert tabs to (4) spaces

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/examples/DatabaseSecurityAlertListByDatabase.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+				"value": [
+					{
+						"id": "/subscriptions/00000000-1111-2222-3333-444444444444/resourceGroups/securityalert-6852/providers/Microsoft.Sql/servers/securityalert-2080/databases/testdb",
+						"name": "default",
+						"type": "Microsoft.Sql/servers/databases/securityAlertPolicies",
+						"location": "Japan East",
+						"kind": "V12",
+						"properties": {
+							"state": "Enabled",
+							"emailAccountAdmins": "true",
```
Comment (sanjaiganesh) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/examples/DatabaseSecurityAlertListByDatabase.json]: tabs to spaces

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/ManagedDatabaseSecurityAlertPolicies.json [hunk 7]
```diff
@@ -159,6 +159,59 @@
           }
         }
       }
+    },
```
Comment (sanjaiganesh) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/ManagedDatabaseSecurityAlertPolicies.json]: any reason for not supporting PATCH   ? (for ex to allow email info to be updated )

**Comment by praries880** [comment]
@t-robalt, Pinging on this.
Are you working to address the issues @sanjaiganesh pointed out in the review?

**Comment by yaakoviyun** [comment]
@praries880  , @sanjaiganesh  - Ron is OOF. Someone else from our team will take it over

**Comment by praries880** [comment]
@t-robalt  @yaakoviyun Are you working on taking care of the review comments?

**Comment by ravbhatnagar** [comment]
review comments from ARM still pending to be addressed.

**Comment by yaakoviyun** [comment]
@praries880 - we will allocate time next week to resolve the issue.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/ManagedDatabaseSecurityAlertPolicies.json (hunk 7 above)
Comment (t-robalt) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/ManagedDatabaseSecurityAlertPolicies.json]: The policy is updated through PUT, so PATCH is not necessary.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ServerSecurityAlertsListByServer.json (hunk 4 above)
Comment (t-robalt) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ServerSecurityAlertsListByServer.json]: Added only location for server.
"kind" is not relevant.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedServerSecurityAlertListByInstance.json (hunk 2 above)
Comment (t-robalt) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedServerSecurityAlertListByInstance.json]: Added only location for server.
"kind" is not relevant.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/ManagedDatabaseSecurityAlertPolicies.json (hunk 7 above)
Comment (KrisBash) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/ManagedDatabaseSecurityAlertPolicies.json]: PATCH is not strictly required for proxy resources.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedServerSecurityAlertListByInstance.json (hunk 2 above)
Comment (KrisBash) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedServerSecurityAlertListByInstance.json]: Sorry for the churn, but `location` is spurious in this case. This is a nested proxy resource that does not have "location" in its definition. Location is not required and is not in the response definition in the swagger spec.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ServerSecurityAlertsListByServer.json (hunk 4 above)
Comment (KrisBash) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ServerSecurityAlertsListByServer.json]: See above. Adding location here makes the example not align to the definition. please remove.

**Comment by t-robalt** [comment]
@KrisBash @sanjaiganesh @praries880  Can someone signoff please?

**Review by KrisBash — APPROVED** [review]
LGTM thanks

**Comment by praries880** [comment]
Closed and reopened the PR to kickoff the CI again.

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5419 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5419/commits/cbf6c5776c914a367a345681903349c0e6166384

**Comment by praries880** [comment]
@sergey-shandar Can you look at the cause of the CI failure here as well?

**Review by praries880 — APPROVED** [review]

**Comment by praries880** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Comment by praries880** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Pull request merged**